main.go

```diff
@@ -47,16 +77,23 @@ func main() {
 			http.Error(w, "Internal server error", http.StatusInternalServerError)
 			slog.Error("Failed parsing darkmode status", "error", err)
 		}
-		systemsToShow := systems
+
+		whichLinks := strings.Split(r.Host, ".")[0]
+		linksToShow, ok := links[whichLinks]
+		if !ok {
+			whichLinks = "systems"
+			linksToShow = links[whichLinks]
+		}
 		if darkmode {
-			systemsToShow = make([]system, 0, len(systems))
-			for _, system := range systems {
-				if !system.Sensitive {
-					systemsToShow = append(systemsToShow, system)
+			linksToShow = make([]link, 0, len(links[whichLinks]))
+			for _, link := range links[whichLinks] {
+				if !link.Sensitive {
+					linksToShow = append(linksToShow, link)
 				}
 			}
```

Comment:
```suggestion
			linksToshow = slices.DeleteFunc(linksToShow, func (l link) bool {
				return l.Sensitive
			})
```
I finally figured it out!